iPhone 6 (Plus) support, App bundle support and showing refunds in the detail view

futuretap opened the pull request:
- Added Launch.storyboard for iPhone 6 (Plus)
- Added missing app icon sizes

Commit:
Corrected Fiscal Calendar thanks to AlessandroPappalardo. Issue #225

Commit:
Fixed the download of payments thanks to AlessandroPappalardo. Issue #241

Commit:
Added the iOS 7 display fix, The reload App Icon fix by @jjochen and the Customer Review download fix by @abuharsky

Commit:
Fixed 10/10/13 unit of sales bug thanks to @ktakayama

Commit:
Fixed download of new app icons

Commit:
Fixed compiler warnings
Updated for App Bundles

Commit:
Totally updated for iOS 8 and iPhone 6's

Commit:
Fixed account deletion

Commit:
Fixed Payments retrieval

Commit:
Fixed Payments YearView footer

futuretap commented:
iPhone 6 (Plus) support, App bundle support and showing refunds in the detail view

Commit:
Show refunds in detail view. Thanks to @futuretap

ddaddy commented:
Thanks @futuretap i've also added your 'Show refunds' code to my branch.

Commit:
zero dollars fix from danielchong

Commit:
Set minimum target to iOS 7 and changed all deprecated methods.
Download last 30 days of reports on first sync, instead of just last 14.

Commit:
Added 64bit support and corrected some view layouts

Commit:
Fix report download failure on iOS 9

sirnacnud commented:
I'm interested in the 'Show refunds' functionality. Any chance that change can be merged in?

ddaddy committed:
Paul Fowler's fix for duplicate apps.

ddaddy committed:
Xcode 7.0.1 update

ddaddy committed:
Added Redownloads view

ddaddy committed:
Only show products in list view if units != 0

ddaddy committed:
Added alert for any errors (Thanks to nicolasgomollon)

nicolasgomollon committed:
Removed RegexKitLite. Disabled PromoCodeOperation.

nicolasgomollon committed:
Converted project to use ARC.

nicolasgomollon committed:
Converted space indentation to tabs.

nicolasgomollon committed:
Updated curly brace position.

nicolasgomollon committed:
Attempted to standardize code styling.

nicolasgomollon committed:
Added Two-Step Verification support.

Also fixed “Auto-Fill Vendor ID” button.

nicolasgomollon committed:
Fixed Payments currency formatting.

nicolasgomollon committed:
Added missing app icons.

nicolasgomollon committed:
Updated launch images and storyboard.

nicolasgomollon committed:
Relabeled Back button.

nicolasgomollon committed:
Fixed sorting by Product ID in Sales and Trends.

nicolasgomollon committed:
Fixed sorting by Product ID in Manage Apps.

nicolasgomollon committed:
Fixed clipped label in sales view.

nicolasgomollon committed:
Increased maximum number of available reports.

nicolasgomollon committed:
Fixed improper year separator.

nicolasgomollon committed:
Improved styling. Locked orientation in some views

nicolasgomollon committed:
Fixed incorrect date labels in sales view.

nicolasgomollon committed:
Fixed small lettering in Sales and Trends.

nicolasgomollon committed:
Removed Boxcar references, since it doesn't work.

nicolasgomollon committed:
Fixed missing screenshot in README.

nicolasgomollon committed:
Added abbreviated weekday to daily reports graph.

nicolasgomollon committed:
Moved login & vendor ID code to background thread.

nicolasgomollon committed:
Fixed review downloads.

- Fixed bug where reviews would not download.
- Fixed bug where reviews would not be fetched from all available pages.
- Removed In-App Purchases from Reviews section, since they can't have reviews and just clutter the screen.
- Removed SMXMLDocument in favor of the native NSXMLParser.

nicolasgomollon committed:
Updated the Yahoo! Finance API URL.

nicolasgomollon committed:
Fixed misplaced orientation method.

nicolasgomollon committed:
Fixed bug where Promo Codes section would be empty

Also fixed bug where new apps without a parent SKU would be saved as whitespace.

nicolasgomollon committed:
Made passcode lock screen less annoying.

nicolasgomollon committed:
Fixed layout issues.

nicolasgomollon committed:
Added Touch ID authentication to unlock app.

nicolasgomollon committed:
Added Total Revenue and Total Sales options.

nicolasgomollon committed:
Fixed missing disclosure indicator in Settings.

nicolasgomollon committed:
Moved Done buttons to the left side.

nicolasgomollon committed:
Fixed missing app icon in About view.

nicolasgomollon committed:
Changed thread behavior of LoginManager.

nicolasgomollon committed:
Fixed minor layout bugs.

nicolasgomollon committed:
Filtered list of apps for Promo Codes.

Also made sure that reviews are only downloaded for apps that are not hidden.

nicolasgomollon committed:
Fixed minor layout bug.

nicolasgomollon committed:
Updated About view.

- Remove references to unused libraries.
- Made app icon @2x and @3x compatible.
- Tweaked app icon corner radius to match iOS 7+.
- Changed web page to use system font (e.g. San Francisco on iOS 9+, Helvetica otherwise).
- Matched web page link color with app tint color.
- Changed app version in web page to use the app's actual version and build numbers.

nicolasgomollon committed:
Fixed bug where Payments status would disappear.

nicolasgomollon committed:
Updated README.

nicolasgomollon committed:
Updated export functionality.

- Added ZIPArchive as a submodule.
- Replaced 37 ZipFile source files with just 11 files for ZIPArchive.

nicolasgomollon committed:
Added MBProgressHUD as a submodule.

- Improved MBProgressHUD used for exporting.
- Updated About view with new licenses.

nicolasgomollon committed:
Removed unused code.

nicolasgomollon committed:
Updated target deployment target to match project.

Also replaced deprecated code with modern equivalents.

nicolasgomollon committed:
Fixed bug related to IAPs in Manage Apps.

nicolasgomollon committed:
Adopted modern Objective-C syntax.

nicolasgomollon committed:
Added SSKeychain as a submodule.

Also updated About view with new license.

nicolasgomollon committed:
Fixed bug with payments saved without a vendor ID.

- Refactored download progress code in ReportDownloadOperation.
- Added a sanity check to ensure that payments are not saved without a proper vendor ID, and an error message to notify the user when the page can't be parsed.
- Added FFGlobalAlertController as a submodule.
- Updated About view with new license.

nicolasgomollon committed:
Updated country list. Improved report detail sort.

nicolasgomollon committed:
Updated UI for iOS 7+, but not quite there yet.

- Changed Deployment Target to iOS 8.2 (developers should be running the latest iOS anyway!)
- Refreshed DashboardAppCell look.
- ColorPickerVC is now CrayonColorPickerVC, and appears as a popover even on iPhone (swipe an app cell to edit color).
- App icons now download at a higher resolution (downloads from iTunes Store if available, otherwise downloads from AppShopper).
- Updated some icons to fit in with iOS 7+.
- Moved images to asset catalog.
- Added @3x assets.
- Replaced flag icons with FlagKit.
- Updated About view with new license.
- Updated ReviewTemplate to use system font.
and more…

nicolasgomollon committed:
Fixed bug with accounts with a single vendor ID.

nicolasgomollon committed:
Tweaked edit product section. Added iPad Pro icon.

nicolasgomollon committed:
Fixed empty toolbar space in PromoCodesVC.

nicolasgomollon committed:
Revamped Customer Reviews section.

- Reviews are now fetched from the (more reliable) iTunes Connect reviews API.
- Reviews are now grouped by version per app.
- Review cells now show some color.
- Delete all existing reviews from your database for best results.

nicolasgomollon committed:
Improved BadgedCell.

- ReviewsVC and PromoCodesVC now include app icons.

nicolasgomollon committed:
Added ability to mark reviews read/unread.

nicolasgomollon committed:
Fixed race condition when downloading reviews.

nicolasgomollon committed:
Fixed ReviewDetailVC layout.

nicolasgomollon committed:
Added country info to ReviewDetailVC.

nicolasgomollon committed:
Updated README. nicolasgomollon/AppSales-Mobile#2

nicolasgomollon committed:
Fixed bug with misplaced “All Apps” label.

nicolasgomollon committed:
Updated README. nicolasgomollon/AppSales-Mobile#3

ddaddy committed:
Added Turkish Lira (TRY) currency.

ddaddy committed:
Updated base currencies.

nicolasgomollon committed:
Updated app build number.

sirnacnud committed:
Fixed AppleFiscalCalendar to extend through the current fiscal period

The AppleFiscalCalendar only covered up to the fifth period of 2016. Now the AppleFiscalCalendar will cover up to one fiscal period past the current fiscal period.

sirnacnud committed:
Replace spaces with tabs

nicolasgomollon committed:
Merge pull request #9 from sirnacnud/fiscal

Fixed AppleFiscalCalendar to extend through the current fiscal period.

nicolasgomollon committed:
Fixed some HTML entities in two-step action sheet.

sirnacnud committed:
Added support to download reviews for Mac Apps

nicolasgomollon committed:
Merge pull request #11 from sirnacnud/reviews

Added support to download reviews for Mac Apps.

nicolasgomollon committed:
Fixed more HTML entities in two-step action sheet.

nicolasgomollon committed:
Updated LoginManager to use new & stable JSON API.

nicolasgomollon committed:
Fixed Payments. nicolasgomollon/AppSales-Mobile#10

- Updated LoginManager to support “Trust this device” option for Two-Step Verification.
- Updated LoginManager to fetch remaining iTunes Connect cookies.
- Fixed “Auto-Fill Vendor ID” button.
- Updated README.
- Updated About view.

sirnacnud committed:
Fix Reviews Not Downloading

kITCBaseURL now includes /WebObjects/iTunesConnect.woa, so it now needs to be removed from the review APIs.

nicolasgomollon committed:
Merge pull request #13 from sirnacnud/reviews

Fixed reviews not downloading.

nicolasgomollon committed:
Updated app build number.

nicolasgomollon committed:
Fixed Payments. nicolasgomollon/AppSales-Mobile#14

- Replaced API call with a functional equivalent.

nicolasgomollon committed:
Added Two-Factor Authentication support.

nicolasgomollon committed:
Moved misplaced files to correct location.

nicolasgomollon committed:
Replaced obsolete SSKeychain with SAMKeychain.

nicolasgomollon committed:
Improved code styling.

nicolasgomollon committed:
Updated to support new Reporter API.

nicolasgomollon committed:
Added new currencies.

Credits: @ddaddy

nicolasgomollon committed:
Fixed Xcode project warnings.

nicolasgomollon committed:
Fixed broken Reporter API.

nicolasgomollon committed:
Fixed bug where cookies needed to be deleted.

futuretap committed:
Payments: fixed for locales with "," as decimalSeparator

such as German;
couldn't parse the payment amount and showed 0€ for all payments;

nicolasgomollon committed:
Merge PR #24 from futuretap/master. Fixes #19.

Payments fixed for locales with "," as decimalSeparator, such as German. Couldn't parse the payment amount, and showed 0€ for all payments as a result.

nicolasgomollon committed:
Updated app build number.

sirnacnud committed:
Remove .gitattributes allowing project.pbxproj to be diffed

sirnacnud committed:
Download and store all payment report data

sirnacnud committed:
Update PaymentsViewController to use new detailed payment data

sirnacnud committed:
Added ability to sort payments by month paid in addition to month earned

The sort button replaces the delete all payments button.

sirnacnud committed:
Fixed crash in Payments when core data object was being used across threads

sirnacnud committed:
Delete and redownload report when pending payment is past the expected payment date

sirnacnud committed:
Make pending payments show up as red

sirnacnud committed:
Convert spaces to tabs

nicolasgomollon committed:
Merge pull request #26 from sirnacnud/payments. Fixes #15.

Improvements to Payments.

nicolasgomollon committed:
Updated app build number. Fixed bugs.

- Fixed cases where there are multiple Vendor IDs by adding up all payments for a month.
- Prevented saving payment entries that are neither paid nor expected.

nicolasgomollon committed:
Fixed #25 with incorrect fiscal dates.

nicolasgomollon committed:
Fixed bug where table does not update.

sirnacnud committed:
Improve ReviewDownloader error handling

Added change to prevent multiple error alerts from stacking up on top of each other. Also added change to allow progress HUD to continue updating while alert is displayed.

sirnacnud committed:
Re-download payment reports which have no payment summaries

Also, fixed to not save a payment report which has no payment summaries.

nicolasgomollon committed:
Merge pull request #27 from sirnacnud/review-errors

Improve ReviewDownloader error handling

nicolasgomollon committed:
Merge pull request #28 from sirnacnud/payment

Re-download payment reports which have no payment summaries

nicolasgomollon committed:
Refined app icon. Added missing app icons.

nicolasgomollon committed:
Updated and added missing passcode assets.

nicolasgomollon committed:
Updated Reviews API. Refactored ReviewDownloader.

nicolasgomollon committed:
Added app version check to About view.

nicolasgomollon committed:
Laid foundation for replying to reviews.

nicolasgomollon committed:
Cleaned up country names.

nicolasgomollon committed:
Revamped Customer Reviews section WIP.

- Reviews are now all listed by default and include review text.
- Reviews can now be filtered using the new Filter button.
- Review cells now indicate if a review has been replied to.

nicolasgomollon committed:
Improved performance of reviews list.

nicolasgomollon committed:
Fixed bug with KVO in ReportDownloadCoordinator.

nicolasgomollon committed:
Refactored reviews downloader.

- Added ability to download reviews for all active apps.
- Refactored ReviewDownloader into ReviewDownloadCoordinator and proper ReviewDownloadOperation.
- Replaced MBProgressHUD with StatusToolbar when downloading reviews.

nicolasgomollon committed:
Fixed crash caused by leftover code.

nicolasgomollon committed:
Replaced Filter icons with editable Fireworks ones

nicolasgomollon committed:
Fixed bug with filters.

nicolasgomollon committed:
Added reviews filter comparators.

Also reworked ReviewFilterViewController interaction mode.

nicolasgomollon committed:
Added update check label to main screen.

- Tweaked logic behind app version check in About view.
- Replaced “Settings” text button with gear icon on main screen.

nicolasgomollon committed:
Fixed minor interface bugs.

- Fixed bug with improperly-sized filter comparator picker.
- Fixed bug when selecting an unread review.
- Fixed bug with review navigation bar title.

nicolasgomollon committed:
Added ReviewListHeaderView.

dave256 committed:
use accesstoken instead of username/password

generate your access token on iTunes Connect website and then enter it in the password field of the AppSales app.

futuretap committed:
Fixed gzipInflate #297

Using this implementation: https://stackoverflow.com/a/17822217/235297
